lib/checks/navigation/identical-links-same-purpose-after.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Skip unless there are more than a single result
+ */
+if (results.length < 2) {
+	return results;
+}
+
+/**
+ * for each result
+ * - get other results with matching accessible name
+ * - check if same purpose is served
+ *  - if not change `result` to `undefined`
+ */
+for (let index = 0; index < results.length; index++) {
+	const { data: nodeData } = results[index];
+	const { name, resource } = nodeData;
+	const identicalNodes = getNodesWithIdenticalAccessibleName(name, index);
+	if (!identicalNodes.length) {
+		continue;
+	}
+	if (!resource) {
+		results[index].result = undefined;
+		continue;
+	}
+	if (!servesSamePurpose(resource, identicalNodes)) {
+		results[index].result = undefined;
+		continue;
+	}
+}
+
+return results;
+
+/**
+ * Get list of nodes from results which match a given accessible name
+ * @method getNodesWithIdenticalAccessibleName
+ * @param {String} expectedName accessible name to be matched
+ * @param {Number} excludeIndex exclude `index` of result, that should not be taken into consideration
+ * @returns {Array<Object>}
+ */
+function getNodesWithIdenticalAccessibleName(expectedName, excludeIndex) {
+	return results.filter((result, index) => {
+		const {
+			data: { name }
+		} = result;
+		return index !== excludeIndex && name === expectedName;
+	});
+}
+
+/**
+ * Check if a given set of nodes have same resource
+ * @method servesSamePurpose
+ * @param {String} expectedResource expected resource
+ * @param {Array<Object>} identicalNodes results where `data.accessibleText` were identical
+ * @returns {Boolean}
+ */
+function servesSamePurpose(expectedResource, identicalNodes) {
+	const { parts: expected } = parseUri(expectedResource);
+	return identicalNodes.every(({ data }) => {
+		const { resource = undefined } = data;
+		if (!resource) {
+			return false;
+		}
+		const { isFile, parts: actual } = parseUri(resource);
+		return actual[isFile ? 'some' : 'every'](part => expected.includes(part));
```

Comment:
This is very unsafe. `parseUri` should not return an array. There's no telling what you're now comparing. What if the hash is the same as the filename or the protocol? Because you turned an object into an array, your data lost all its integrity. This `? 'some' : 'every'` thing also makes me very nervous. So if the protocol is the same it'll pass? Why?

Please rethink this approach.